examples/cli.rs

```diff
@@ -0,0 +1,25 @@
+use std::env;
+
+fn main() {
+    let mut args = env::args();
+    let argv0 = args.next()
+        .expect("First argument is always present");
+
+    if args.size_hint().1 == Some(0) {
+        eprintln!("No URIs were given on the command line.");
+        eprintln!("Try running this as `{} http://example.com:1234/hello ../../path`", argv0);
+    }
+
+    for uri in args {
+        let parsed = uriparse::URIReference::try_from(uri.as_str());
+        println!("<{}>: {:#?}", uri, parsed);
+
+        if let Ok(parsed) = parsed {
```

Comment:
Is this part needed? There are some URIs that necessarily won't roundtrip but it's not necessarily an implementation problem. For example, `http://000.000.000.000/` will format as `http://0.0.0.0/` and there's not much to do about this. Another example is the trailing slash, `http://example.com` will format as `http://example.com/` as authorities always have such a trailing slash.